### PR TITLE
Separating EV departure and arrival messages

### LINKED ISSUE #77: Separate EV departure and arrival message exchanges
Due to new protocol requirements, EV departures and arrivals need to be separated.

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #77 

Also:
- Adding ScalaDocs and refactoring messages
- Removing message builders
- Properly handling InterruptedException
- Abstracting message receiving and casting to a generic method in ExtEvData

### CODE REVIEW

File: src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java [hunk 1]
```diff
@@ -54,52 +55,63 @@ public Map<UUID, Integer> requestAvailablePublicEvcs() {
...
+      if (evMessage.getClass().equals(ProvideCurrentPrices.class)) {
+        final ProvideCurrentPrices provideCurrentPrices = (ProvideCurrentPrices) evMessage;
+        return provideCurrentPrices.prices();
       }
     } catch (InterruptedException ie) {
       Thread.currentThread().interrupt();
     }
 
-    return new ArrayList<>();
+    return new HashMap<>();
```
Comment (t-ober) [inline src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java:78]: Should this ever happen? 
Wouldn't this mean the message in the queue we got would be of a false kind? 
In which case I would rather throw an exc

File: src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java [hunk 2]
```diff
@@ -54,52 +55,63 @@ public Map<UUID, Integer> requestAvailablePublicEvcs() {
...
+      if (evMessage.getClass().equals(ProvideDepartingEvs.class)) {
+        final ProvideDepartingEvs departedEvsResponse = (ProvideDepartingEvs) evMessage;
+        return departedEvsResponse.departedEvs();
       }
     } catch (InterruptedException ie) {
       Thread.currentThread().interrupt();
     }
 
-    return new HashMap<>();
+    return new ArrayList<>();
```
Comment (t-ober) [inline src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java:104]: s.o.

**Review by t-ober — CHANGES_REQUESTED** [review]
That was quick ! Thanks 🙂 

Just one question

File: src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java (hunk 1 above)
Comment (sebastian-peter) [reply src/main/java/edu/ie3/simona/api/data/ev/ExtEvData.java:78]: You're right, it seems like [best practice](https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java) is to let the exception escalate. This means you'll have to handle it in MobSim, probably by exiting the simulation :)

**Review by t-ober — CHANGES_REQUESTED** [review]
Thanks a lot - I like the tests!

**Review by t-ober — APPROVED** [review]

**Pull request merged**